### PR TITLE
refactor!: Remove task category

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Companion PR

- https://github.com/Substra/substra-backend/pull/528

## Description

Remove category from tasks.

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: server/standalone/migration/000050_remove-task-category.up.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        t.algo_key AS algo_key,
+        COALESCE(p.parent_task_keys, '[]'::jsonb) AS parent_task_keys
+    FROM compute_tasks t
+             LEFT JOIN (
+        SELECT child_task_key, JSONB_AGG(parent_task_key) AS parent_task_keys
+        FROM compute_task_parents
+        GROUP BY child_task_key
+    ) p ON p.child_task_key = t.key;
+
+    ALTER TABLE compute_tasks DROP COLUMN category;
```
Comment (thibowk) [inline server/standalone/migration/000050_remove-task-category.up.sql:26]: I think you also need to remove the `compute_task_categories` table.

https://github.com/Substra/orchestrator/blob/main/server/standalone/migration/000015_remove_compute_task_asset_column.up.sql#L6

File: server/standalone/migration/000050_remove-task-category.up.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+
```
Comment (thibowk) [inline server/standalone/migration/000050_remove-task-category.up.sql:2]: Maybe remove those empty lines.

File: server/standalone/migration/000050_remove-task-category.up.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
```
Comment (thibowk) [inline server/standalone/migration/000050_remove-task-category.up.sql:1]: To be consistent with previous migration files, I guess you should only use `_` and not `-` in the name of the file.

**Review by thibowk — COMMENTED** [review]
Thanks for these welcome changes!

File: server/standalone/migration/000050_remove-task-category.up.sql (hunk 1 above)
Comment (guilhem-barthes) [reply server/standalone/migration/000050_remove-task-category.up.sql:26]: Good point!

Comment (guilhem-barthes) [reply server/standalone/migration/000050_remove-task-category.up.sql]: Addressed in https://github.com/Substra/orchestrator/commit/ab77e90829b75b0eb28d3caa174827ba3805c134

**Review by AurelienGasser — APPROVED** [review]
LGTM

**Comment by AurelienGasser** [comment]
Note that the PR in this migration conflicts with https://github.com/Substra/orchestrator/pull/46

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                                         Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: substra-hlf-k8s, orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>                 Comma-separated list of tests to run. Valid options: sdk,substrafl,frontend or NONE. (default: "sdk")
  --benchmarks-to-run, --benchmarks <benchmarks-to-run>  Comma-separated list of workflows tests to run. Valid options: mnist, camelyon or NONE. (default: "NONE")
  --orchestrator-mode, --mode <orchestrator-mode>        Comma-separated list of orchestrator modes to run tests for. Valid options: standalone,distributed (default: "standalone,distributed")
  -h, --help                                             display help for command
```

**Comment by oleobal** [comment]
/e2e --refs substra-backend=feat/remove_task_category,substra-ci=fix/revert-gcloud-version

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: server/standalone/dbal/computetask_test.go [hunk 4]
```diff
@@ -108,7 +106,7 @@ func TestTaskFilterToQuery(t *testing.T) {
 		"empty":         {&asset.TaskQueryFilter{}, "", nil},
 		"single filter": {&asset.TaskQueryFilter{Worker: "myorganization"}, "worker = $1", []interface{}{"myorganization"}},
 		"two filter":    {&asset.TaskQueryFilter{Worker: "myorganization", Status: asset.ComputeTaskStatus_STATUS_DONE}, "worker = $1 AND status = $2", []interface{}{"myorganization", asset.ComputeTaskStatus_STATUS_DONE.String()}},
-		"three filter":  {&asset.TaskQueryFilter{Worker: "myorganization", Status: asset.ComputeTaskStatus_STATUS_DONE, Category: asset.ComputeTaskCategory_TASK_TRAIN}, "worker = $1 AND status = $2 AND category = $3", []interface{}{"myorganization", asset.ComputeTaskStatus_STATUS_DONE.String(), asset.ComputeTaskCategory_TASK_TRAIN.String()}},
+		"three filter":  {&asset.TaskQueryFilter{Worker: "myorganization", Status: asset.ComputeTaskStatus_STATUS_DONE, AlgoKey: "c14b4e4b-132b-4321-972d-b1506f82e5cd"}, "worker = $1 AND status = $2 AND algo_key = $3", []interface{}{"myorganization", asset.ComputeTaskStatus_STATUS_DONE.String(), "c14b4e4b-132b-4321-972d-b1506f82e5cd"}},
```
Comment (ThibaultFy) [inline server/standalone/dbal/computetask_test.go:109]: Could we change the algo key to "test-key" for instance to make the test easier to read ?

File: server/standalone/migration/000052_remove-task-category.up.sql [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+             LEFT JOIN (
+        SELECT child_task_key, JSONB_AGG(parent_task_key) AS parent_task_keys
+        FROM compute_task_parents
+        GROUP BY child_task_key
+    ) p ON p.child_task_key = t.key;
+
+    ALTER TABLE compute_tasks DROP COLUMN category;
+    DROP TABLE compute_task_categories;
+
+    UPDATE events SET asset = asset #- '{category}'
```
Comment (ThibaultFy) [inline server/standalone/migration/000052_remove-task-category.up.sql:27]:
```suggestion
    UPDATE events SET asset = asset - '{category}'
```
Check without the # ?

File: docs/schemas/entities.mmd [hunk 6]
```diff
@@ -1,9 +1,7 @@
 erDiagram
     DataManager }|--o{ DataSample : opens
     ComputeTask }o--|| ComputePlan : "belongs to"
-    ComputeTask {
-        ComputeTaskCategory category
-    }
+    ComputeTask {}
```
Comment (ThibaultFy) [inline docs/schemas/entities.mmd:4]: Can we remove this line ?

**Review by ThibaultFy — COMMENTED** [review]
Thanks for the PR

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: server/standalone/dbal/computetask_test.go (hunk 4 above)
Comment (guilhem-barthes) [reply server/standalone/dbal/computetask_test.go]: Tried, reached the following error in distributed tests
```json
{
   "event": "Internal Server Error: /data_sample/",
   "exc_info": [
      "<class 'django.core.exceptions.ValidationError'>",
      "ValidationError(['\u201cfake_key\u201d is not a valid UUID.'])",
      "<traceback object at 0x7f6ab66aeb80>"
   ],
   "timestamp": "2022-12-23T14:23:23.214535Z",
   "logger": "django.request",
   "level": "error"
}
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — APPROVED** [review]
LGTM Thanks !!

**Pull request merged**